src/unix/os390-syscalls.c

```diff
@@ -332,3 +332,48 @@ char* mkdtemp(char* path) {
 
   return path;
 }
+
+
+ssize_t os390_readlink(const char* path, char* buf, size_t len) {
+  ssize_t rlen;
+  char* delimiter;
+  char old_delim;
+  char tmpbuf[len + 1];
+  char realpathstr[PATH_MAX];
+
+  rlen = readlink(path, tmpbuf, len);
+  if (rlen < 0)
+    return rlen;
+
+  if (rlen < 3 || strncmp("/$", tmpbuf, 2) != 0) {
+    /* Straightforward readlink */
+    memcpy(buf, tmpbuf, rlen);
+    return rlen;
+  }
+
+  /*
+   * There is a parmlib variable at the beginning
+   * which needs interpretation
+   */
+  tmpbuf[rlen] = '\0';
+  delimiter = strchr(tmpbuf + 2, '/'); 
+  if (delimiter == NULL)
+    /* No slash at the end */
+    delimiter = strchr(tmpbuf + 2, '\0');
+
+  /* Read real path of the variable */
+  old_delim = *delimiter;
+  *delimiter = '\0';
+  if (realpath(tmpbuf, realpathstr) == NULL)
+    return -1;
+
+  /* Reset the delimiter and fill up the buffer */
+  *delimiter = old_delim;
+  rlen = snprintf(buf, len, "%s%s", realpathstr, delimiter);
```

Comment:
Apologies if I'm missing something obvious but thought that was the case because of [this](https://github.com/libuv/libuv/blob/v1.x/src/unix/fs.c#L432):.